### ISSUE TITLE
Environment.Required_ and others are gone

### ISSUE THREAD

**Issue by mexx** [freya-fs/freya#171]
c50e33edabf5ee785a1c0e269ceb508866b8f7c6 removed them without providing `Obsolete` warning

**Comment by kolektiv** [comment]
Yes, probably worth me going through and making sure that obsolete variants exist where possible. I have been trying to do so, but these things might have happened before I adopted a "backwards compatible until 4.0" policy!
